### PR TITLE
Swift5 models improvements

### PR DESCRIPTION
**Pull request by aymanbagabas**
- Add the option to use `final class` instead of `struct`
- Always include `CodingKeys`
- Implement equals and `hash` funcs
- Encode `null` values
- Add `useClasses` to urlsessionLibrary config
- Add a required nullable test case to 2_0/swift/petstore*.yaml

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by aymanbagabas** [comment]
```
Dependencies ready!
I need the list of published packages to verify your dependencies, so I tried to
fetch:

    https://package.elm-lang.org/all-packages

But it came back as 502 Bad Gateway
```
image
Good times!

**Comment by 4brunu** [comment]
I'm a bit tired today, so tomorrow I will review this more carefully, but some initial thoughts, is that I thinks we shouldn't create the equals and hash functions on structs, because they are given automatically for free.

**Comment by aymanbagabas** [comment]
> I thinks we shouldn't create the equals and hash functions on structs, because they are given automatically for free.

You're correct, `struct`s have a default implementation of `func ==` and `hash(to:)`. PR updated.

File: modules/openapi-generator/src/main/resources/swift5/modelObject.mustache [hunk 1]
```diff
@@ -30,12 +30,17 @@
...
 
-    {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}} init({{#allVars}}{{name}}: {{{datatypeWithEnum}}}{{#required}}{{#isNullable}}?{{/isNullable}}{{/required}}{{^required}}?{{/required}}{{#defaultValue}} = {{{defaultValue}}}{{/defaultValue}}{{^defaultValue}}{{^required}} = nil{{/required}}{{/defaultValue}}{{^-last}}, {{/-last}}{{/allVars}}) {
+    {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}} init({{#allVars}}{{{name}}}: {{{datatypeWithEnum}}}{{#required}}{{#isNullable}}?{{/isNullable}}{{/required}}{{^required}}?{{/required}}{{#defaultValue}} = {{{defaultValue}}}{{/defaultValue}}{{^defaultValue}}{{^required}} = nil{{/required}}{{/defaultValue}}{{^-last}}, {{/-last}}{{/allVars}}) {
         {{#allVars}}
-        self.{{name}} = {{name}}
+        self.{{{name}}} = {{{name}}}
         {{/allVars}}
     }
 {{/hasVars}}
+    {{#nonPublicApi}}internal{{/nonPublicApi}}{{^nonPublicApi}}public{{/nonPublicApi}} enum CodingKeys: {{#hasVars}}String, {{/hasVars}}CodingKey, CaseIterable {
```
Comment (4brunu) [inline modules/openapi-generator/src/main/resources/swift5/modelObject.mustache:39]: @aymanbagabas some small nit picks, could you please create a new line between line 38 and 39?

**Comment by 4brunu** [comment]
I think that maybe you forgot to update the sample projects, because the structs are still implementing equals and hash functions

**Comment by wing328** [comment]
Please resolve the merge conflicts when you've time.

**Comment by wing328** [comment]
> Add a required nullable test case to 2_0/swift/petstore*.yaml


I think you will need to do it in the OpenAPI 3.0 spec instead as nullable is only officially support in 3.0 but not 2.0

**Comment by aymanbagabas** [comment]
> I think you will need to do it in the OpenAPI 3.0 spec instead as nullable is only officially support in 3.0 but not 2.0

I think this is fine since OAS 2.0 `x-nullable` gets mapped to OAS 3.0 `nullable`, right? Plus currently, there isn't a `3_0/swift/petstore*.yaml` specs file.

**Comment by wing328** [comment]
> I think this is fine since OAS 2.0 x-nullable gets mapped to OAS 3.0 nullable, right? Plus currently, there isn't a 3_0/swift/petstore*.yaml specs file.

We are actually move away from 2.0 spec and use mainly 3.0 for samples. Don't worry about it now. I'll do the migration later.

Please resolve the merge conflicts when you've time

**Pull request merged**